tests/GridFS/ReadableStreamFunctionalTest.php

```diff
@@ -190,31 +192,108 @@ public function testSeekOutOfRange()
+    }
+
+    public function providePreviousChunkSeekOffsetAndBytes()
+    {
+        return [
+            [0, 4, 'abcd'],
+            [2, 4, 'cdef'],
+            [4, 4, 'efgh'],
+            [6, 4, 'ghij'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideSameChunkSeekOffsetAndBytes
+     */
+    public function testSeekSameChunk($offset, $length, $expectedBytes)
     {
         $fileDocument = $this->collectionWrapper->findFileById('length-10');
         $stream = new ReadableStream($this->collectionWrapper, $fileDocument);
 
-        $stream->readBytes(1);
-        $stream->seek(5);
-        $stream->seek(2);
-        $stream->readBytes(1);
+        // Read to initialize and advance the chunk iterator to the middle chunk
+        $this->assertSame('abcdef', $stream->readBytes(6));
+
+        $commands = [];
+
+        (new CommandObserver)->observe(
+            function() use ($stream, $offset, $length, $expectedBytes) {
+                $stream->seek($offset);
+                $this->assertSame($expectedBytes, $stream->readBytes($length));
+            },
+            function(stdClass $command) use (&$commands) {
+                $commands[] = key((array) $command);
+            }
+        );
+
+        $this->assertSame([], $commands);
     }
 
-    public function testSeekSubsequentChunk()
+    public function provideSameChunkSeekOffsetAndBytes()
+    {
+        return [
+            [4, 4, 'efgh'],
+            [6, 4, 'ghij'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideSubsequentChunkSeekOffsetAndBytes
+     */
+    public function testSeekSubsequentChunk($offset, $length, $expectedBytes)
     {
         $fileDocument = $this->collectionWrapper->findFileById('length-10');
+        $stream = new ReadableStream($this->collectionWrapper, $fileDocument);
 
-        $observer = $this->getMockBuilder(ReadableStream::class)
-                         ->setConstructorArgs(array($this->collectionWrapper, $fileDocument))
-                         ->getMock();
+        // Read to initialize the chunk iterator to the first chunk
+        $this->assertSame('a', $stream->readBytes(1));
 
-        $observer->expects($this->never())
-                 ->method('initChunksIterator');
+        $commands = [];
 
-        $observer->readBytes(1);
-        $observer->seek(5);
-        $observer->seek(2);
-        $observer->readBytes(1);
+        (new CommandObserver)->observe(
+            function() use ($stream, $offset, $length, $expectedBytes) {
+                $stream->seek($offset);
+                $this->assertSame($expectedBytes, $stream->readBytes($length));
+            },
+            function(stdClass $command) use (&$commands) {
+                $commands[] = key((array) $command);
+            }
+        );
+
+        $this->assertSame([], $commands);
+    }
+
+    public function provideSubsequentChunkSeekOffsetAndBytes()
+    {
+        return [
+            [4, 4, 'efgh'],
```

Comment:
This test case should expose a bug in https://github.com/mongodb/mongo-php-library/pull/491. I'll follow up in that PR with a comment explaining why.